addon/state-for.js

```diff
@@ -79,8 +79,18 @@ function createStateFor(context, stateName, owner) {
   }
 
   if (EmberObject.detect(StateFactoryClass)) {
-    return StateFactoryClass.create(defaultState);
+    return createClassInstance(StateFactoryClass, defaultState, owner);
   }
 
-  return EmberObject.create(StateFactoryClass);
+  return createClassInstance(EmberObject, StateFactoryClass, owner);
+}
+
+function createClassInstance(cls, props, owner) {
+  if (typeof owner.ownerInjection === 'function') {
+    // ember >= 2.3.0
+    return cls.create(owner.ownerInjection(), props)
+  }
+
+  // ember < 2.3.0
+  return cls.extend({ container: owner.__container__ }).create(props)
```

Comment:
The overall concept I am ok with. I don't think we should handle the case before < 2.3.0 personally. I would rather us just do:

```js
if (EmberObject.detect(StateFactoryClass)) {
	return StateFactoryClass.create(
		owner.ownerInjection(),
		defaultState
	);
}

return EmberObject.create(owner.ownerInjection(), StateFactoryClass);
```

Reply:
Done!